### PR TITLE
PLANNER-2280, PLANNER-2282: Generic moves for list variable

### PR DESCRIPTION
**Pull request by yurloc**
### JIRA
- https://issues.redhat.com/browse/PLANNER-2280
- https://issues.redhat.com/browse/PLANNER-2282

## Done
- Change and swap moves.
- Move selectors.
  - Temporary implementation for original/shuffled selection order.
  - Draft implementation of random selection order. Performs well but the implementation is not final as some workarounds are used.
- GenuineVariableDescriptor subclassed by ListVariableDescriptor and BasicVariableDescriptor.

## To do (probably out of scope of this PR)
- Original selection order.
- Generate moves using a ValueSelector instead of an EntitySelector (iterate over tasks not employees) to improve selection fairness.
- Replace custom shadow variable listeners for index (in list variable) and inverse relation (the entity owning the list variable) with generic listeners.

## Feature development methodology
- Long-lived feature branch `list-variable-task-assigning` that will incrementally receive PRs until the feature is ready for master.
- Duplicated `taskassigning` example.
  - Allows to run chained and list variable versions on the same branch.

OptaPlanner examples _029

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|apps|examples] native</b>
</details>

### CODE REVIEW

**Comment by Christopher-Chianelli** [comment]
TODO:

- Add `@PlanningCollectionVariable` to`PLANNING_ENTITY_FIELD_ANNOTATIONS` it optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/.../DotNames.java. Also to BeanDefiningAnnotations once it has strengthFactoryClass etc.

- If SolutionCloner/MemberAccessors are touched, let me know (since it'll probably require some changes to the Gizmo code)

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/descriptor/BasicVariableDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+package org.optaplanner.core.impl.domain.variable.descriptor;
+
+import org.optaplanner.core.api.domain.variable.PlanningVariable;
+import org.optaplanner.core.api.domain.variable.PlanningVariableGraphType;
+import org.optaplanner.core.impl.domain.common.accessor.MemberAccessor;
+import org.optaplanner.core.impl.domain.entity.descriptor.EntityDescriptor;
+import org.optaplanner.core.impl.domain.policy.DescriptorPolicy;
+
+// TODO naming
```
Comment (yurloc) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/descriptor/BasicVariableDescriptor.java:25]: SimpleVariable?

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/descriptor/BasicVariableDescriptor.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+public class BasicVariableDescriptor<Solution_> extends GenuineVariableDescriptor<Solution_> {
```
Comment (yurloc) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/descriptor/BasicVariableDescriptor.java:26]: SimpleVariableDescriptor?

**Comment by yurloc** [comment]
Jenkins retest this.

**Comment by yurloc** [comment]
Jenkins run tests.

**Comment by mareknovotny** [comment]
Jenkins retest this

**Pull request closed without merging**